components/sbm-core/src/test/java/org/springframework/sbm/project/resource/TestProjectContext.java

```diff
@@ -575,18 +568,23 @@ private Integer getOrder(ProjectResourceWrapper l1) {
         @NotNull
         private ProjectContextInitializer createProjectContextInitializer() {
             AtomicReference<ProjectContextInitializer> projectContextInitializerRef = new AtomicReference<>();
-            if(beanFactory != null) {
+            if (beanFactory != null) {
                 ProjectContextInitializer bean = beanFactory.getBean(ProjectContextInitializer.class);
                 projectContextInitializerRef.set(bean);
                 executionContext = beanFactory.getBean(ExecutionContext.class);
             } else {
-                Map<Class<?>, Object> replacedBean = new HashMap<>();
-                if(sbmApplicationProperties != null) {
-                    replacedBean.put(SbmApplicationProperties.class, sbmApplicationProperties);
+                Map<Class<?>, Object> replacedBeans = new HashMap<>();
+                if (sbmApplicationProperties != null) {
+                    replacedBeans.put(SbmApplicationProperties.class, sbmApplicationProperties);
+                }
+
+                if (eventPublisher != null) {
+                    replacedBeans.put(ApplicationEventPublisher.class, eventPublisher);
```

Comment:
This doesn't work as the bean implementing `ApplicationEventPublisher` is the `ApplicationContext` itself.
The whole "bean replacement" approach needs a rehaul.